src/creature_tracker.h

```diff
@@ -24,7 +24,7 @@ class Creature_tracker
         {
             public:
                 bool operator()( const std::weak_ptr<monster> &lhs, const std::weak_ptr<monster> &rhs ) const {
-                    return lhs.lock().get() == rhs.lock().get();
+                    return lhs.lock().get() < rhs.lock().get();
```

Comment:
If we want to get really fastidious about defined behaviour, you ought to use `std::less` to compare pointers, `operator<` is only valid to compare pointers from the same allocation.  But this is only a problem in practice on 16-bit architectures.